frontends/verilog/verilog_parser.y

```diff
@@ -715,6 +715,16 @@ logic_type:
 		astbuf3->range_left = 31;
 		astbuf3->range_right = 0;
 		astbuf3->is_signed = true;
+	} |
+	TOK_INT {
+		astbuf3->is_reg = true;
+		astbuf3->is_signed = true;
+		addRange(astbuf3);
+	} |
+	TOK_SHORTINT {
+		astbuf3->is_reg = true;
```

Comment:
I don't think these should mark `is_reg` because they will already be marked `is_logic` on lines 698 or 701. I think setting `is_reg` for these types is hiding a latent issue.

I am currently getting a warning for 
```SystemVerilog
for (integer x = 0; x < 1; x++)
```
but not for
```SystemVerilog
for (int x = 0; x < 1; x++)
```
nor for
```SystemVerilog
for (shortint x = 0; x < 1; x++)
```

If I remove the errant `is_reg` lines here, all of the above produce warnings, whereas none of the below produce warnings _with or without_ setting `is_reg`.
```SystemVerilog
integer x;
for (x = 0; x < 1; x++)
```
```SystemVerilog
int x;
for (x = 0; x < 1; x++)
```
```SystemVerilog
shortint x;
for (x = 0; x < 1; x++)
```

In sorting out these warnings, I think we would benefit from test cases for these, likely making use of `logger -expect-no-warnings`.

Reply:
Fixed. I also created new test case with ``logger -expect-no-warnings``.